src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.cs

```diff
@@ -73,6 +73,21 @@ private static bool GetIsWindowsSubsystemForLinux()
             return false;
         }
 
+        static Lazy<bool> s_largeArrayIsNotSupported = new Lazy<bool>(() =>
+        {
+            try
+            {
+                var tmp = new byte[int.MaxValue];
+                return tmp == null;
```

Comment:
This should to be a black box operation that the optimizations are not able to eliminate. `tmp == null` can be optimized out easily.

FWIW, RyuJIT does some of these optimizations today already. For example, if you have:

```csharp
class Test
{
    static public bool Method()
    {
        Test t = new Test();
        return (t != null);
    }
}
```

It will get optimized to just `return true`.